### PR TITLE
update script for colony building entries

### PR DESCRIPTION
**Pull request by dbenage-cx**
Missed free standing python utility: https://github.com/freeorion/freeorion/pull/559#issuecomment-201966609

### CODE REVIEW

File: default/scripting/buildings/colonies/col_bld_gen.py [hunk 1]
```diff
@@ -211,19 +212,22 @@
     sp_id = species[0]
     sp_name = sp_id.split("_", 1)[1]
     sp_desc_name = species[1]
+    sp_tags = "\""+sp_id+"\""
     sp_graphic = species[2]
     sp_filename = sp_id + ".focs.txt"
     with open(os.path.join(outpath, sp_filename), "w") as f:
         if sp_id == "SP_EXOBOT":
-            f.write(t_main.substitute(id=sp_id, name=sp_name, graphic=sp_graphic, cost=70, time=5,
+            sp_tags += " \"CTRL_ALWAYS_REPORT\""
```
Comment (Cjkjvfnby) [inline default/scripting/buildings/colonies/col_bld_gen.py]: Use single quotes to wrap double quotes
`" \"CTRL_ALWAYS_REPORT\""` => `' "CTRL_ALWAYS_REPORT"'`

**Comment by adrianbroher** [comment]
@dbenage-cx
Sorry, what are you talking about and doing here?

Neither the PR body nor the commits give a reference what are you doing here, why this PR is needed or if this PR is a continuation of another PR/commit.

Also while you updated the script it's more interesting that you've added tag support to said script.  These infos are essential to browse through the history with just looking at the commit message.

Please update the commit message accordingly.

**Comment by dbenage-cx** [comment]
rebased

@adrianbroher apologies, will make sure to have more distinct commit messages

File: default/scripting/buildings/colonies/col_bld_gen.py [hunk 2]
```diff
@@ -211,19 +212,22 @@
     sp_id = species[0]
     sp_name = sp_id.split("_", 1)[1]
     sp_desc_name = species[1]
+    sp_tags = '"'+sp_id+'"'
```
Comment (adrianbroher) [inline default/scripting/buildings/colonies/col_bld_gen.py]: We try to follow PEP 0008, so please add spaces around the `+` operator.

[PEP 0008 - Other recommendations](https://www.python.org/dev/peps/pep-0008/#other-recommendations)

File: default/scripting/buildings/colonies/col_bld_gen.py [hunk 3]
```diff
@@ -211,19 +212,22 @@
...
     sp_graphic = species[2]
     sp_filename = sp_id + ".focs.txt"
     with open(os.path.join(outpath, sp_filename), "w") as f:
         if sp_id == "SP_EXOBOT":
-            f.write(t_main.substitute(id=sp_id, name=sp_name, graphic=sp_graphic, cost=70, time=5,
+            sp_tags += ' "CTRL_ALWAYS_REPORT"'
+            f.write(t_main.substitute(id=sp_id, name=sp_name, tags=sp_tags, graphic=sp_graphic, cost=70, time=5,
```
Comment (adrianbroher) [inline default/scripting/buildings/colonies/col_bld_gen.py]: ~~Same goes for lines you change. When there is a style violation in the line you change try to fix it but no more.  So in this case please add spaces around the `=` operator.~~

File: default/scripting/buildings/colonies/col_bld_gen.py [hunk 4]
```diff
@@ -211,19 +212,22 @@
...
                     species_condition=r"// no existing Exobot colony required!") + "\n\n")
         elif sp_id == "SP_BANFORO" or sp_id == "SP_KILANDOW" or sp_id == "SP_MISIORLA":
             this_time_factor = time_factor.get(sp_id, "1.0")
-            f.write(t_main.substitute(id=sp_id, name=sp_name, graphic=sp_graphic, cost=50,
+            sp_tags += ' "CTRL_EXTINCT"'
+            f.write(t_main.substitute(id=sp_id, name=sp_name, tags=sp_tags, graphic=sp_graphic, cost=50,
```
Comment (adrianbroher) [inline default/scripting/buildings/colonies/col_bld_gen.py]: ~~Also missing spaces around `=`.~~

File: default/scripting/buildings/colonies/col_bld_gen.py [hunk 5]
```diff
@@ -211,19 +212,22 @@
...
                                       time=t_buildtime_extinct.substitute(id=sp_id, name=sp_name, t_factor=this_time_factor),
                                       species_condition=t_species_condition_extinct.substitute(id=sp_id, name=sp_name)) + "\n\n")
         else:
             this_time_factor = time_factor.get(sp_id, "1.0")
-            f.write(t_main.substitute(id=sp_id, name=sp_name, graphic=sp_graphic, cost=50,
+            f.write(t_main.substitute(id=sp_id, name=sp_name, tags=sp_tags, graphic=sp_graphic, cost=50,
```
Comment (adrianbroher) [inline default/scripting/buildings/colonies/col_bld_gen.py]: ~~Also missing spaces around `=`.~~

**Comment by adrianbroher** [comment]
The PR looks good.

I tried the script and the tags were added with the right syntax.  Is there a particular reason why this PR doesn't contain another commit with the updated colony scripts?

Also about the style remark: It's okay if you just want to change the offending line.  But if you want to you can add yet another (optional) commit to fix further style violations.

File: default/scripting/buildings/colonies/col_bld_gen.py (hunk 3 above)
Comment (adrianbroher) [inline default/scripting/buildings/colonies/col_bld_gen.py]: Nevermind, that notation is already correct.

**Comment by dbenage-cx** [comment]
pep8 lint only fails now for long lines.
As the templates would be hard to read if broken before col 79, I assume this is ok? (extending to 99)

updated building entries with generated output.

**Comment by Cjkjvfnby** [comment]
I use 120 chars for code instead of 79.

Long lines in template are OK.

**Comment by Cjkjvfnby** [comment]
I fix long calls in other way: https://github.com/freeorion/freeorion/pull/569

**Pull request merged**